press/api/developer/marketplace.py

```diff
@@ -86,6 +88,53 @@ def get_login_url(self):
 			f"/api/method/press.api.marketplace.login_via_token?token={token}&team={team}&site={self.app_subscription_doc.site}"
 		)
 
+	def get_subscriptions(self):
+		from press.api.marketplace import get_plans_for_app
+
+		team = self.app_subscription_doc.team
+		currency, address = frappe.db.get_value("Team", team, ["currency", "billing_address"])
+		response = {"currency": currency, "address": True if address else False}
+		response["subscriptions"] = [
+			s.update(
+				{
+					"available_plans": get_plans_for_app(s["app"]),
+					**frappe.db.get_value(
+						"Marketplace App", s["app"], ["title", "image"], as_dict=True
+					),
+				}
+			)
+			for s in frappe.get_all(
+				"Marketplace App Subscription",
+				filters={"team": self.app_subscription_doc.team, "status": "Active"},
+				fields=["name", "app", "site", "plan"],
+			)
+		]
+
+		return response
+
+	def update_billing_info(self, data: Dict) -> str:
+		self.login_as_team()
+		team_doc = frappe.get_doc("Team", self.app_subscription_doc.team)
+		team_doc.update_billing_details(data)
+
+		return "success"
+
+	def saas_payment(self, data):
+		self.login_as_team()
+		return prepaid_saas_payment(
+			self.app_subscription_name,
+			self.app_subscription_doc.app,
+			self.app_subscription_doc.site,
+			data["new_plan"]["name"],
+			data["total"],
+			data["total"],
+			data["billing"],
+			False,
+		)
+
+	def login_as_team(self):
```

Comment:
@adityahase @ankush Do you think there's a better way to do this? Or this looks good enough?

ref: update_billing_info and prepaid_saas_payment does a lot of restricted stuff which need user to be logged in